File: utils/extract_embedding_cluster.py
import scanpy as sc
import numpy as np
import pandas as pd
import scipy.io
import argparse
import os
from collections import Counter

import sys
sys.path.append("../")
from core.utils import subset_by_column,subset_by_cell_feature


parser=argparse.ArgumentParser()
parser.add_argument("-s","--subset",nargs="+",type=str,default=None)
parser.add_argument("-o","--outdir",type=str,default=None)
parser.add_argument("-d","--data",type=str,default=None,help="adata object")
parser.add_argument("-c","--column",type=str,default="leiden")
parser.add_argument("-v","--invert",action="store_true",default=False)

args=parser.parse_args()
if not os.path.exists(args.outdir):
   os.makedirs(args.outdir)


print("### Laoding data")
adata=sc.read_h5ad(args.data)

if args.subset is not None:
    print("*** subset")
    print("Subset data by {}".format(args.column))
    adata=subset_by_column(adata,args.subset,args.column,invert=args.invert)

metadata=adata.obs.copy()
metadata["barcode"]=metadata.index
print(metadata.head())

assert args.column in metadata.columns
print("Export cluster")
df_cluster=metadata[["barcode",args.column]]
df_cluster.to_csv(os.path.join(args.outdir,"cluster.csv"),sep=",",index=False)

print("Export embeding")
############### umap
umap_mat=adata.obsm["X_umap"]
umap_mat=pd.DataFrame(umap_mat,columns=["UMAP_1","UMAP_2"],index=adata.obs_names)
umap_mat["Barcode"]=umap_mat.index

df_umap=umap_mat.iloc[:,[2,0,1]]
df_umap.to_csv(os.path.join(args.outdir,"umap_projection.csv"),sep=",",index=False)
################ tsne
tsne_mat=adata.obsm["X_tsne"]
tsne_mat=pd.DataFrame(tsne_mat,columns=["tSNE_1","tSNE_2"],index=adata.obs_names)
tsne_mat["Barcode"]=tsne_mat.index

df_tsne=tsne_mat.iloc[:,[2,0,1]]
df_tsne.to_csv(os.path.join(args.outdir,"tsne_projection.csv"),sep=",",index=False)

#df_ident_cluster=metadata[["idents",args.column]]
#df_ident_cluster=metadata[["Sample",args.column]]
#df_ident_cluster.to_csv(os.path.join(args.outdir,"sample_cluster.csv"),sep=",",index=False)

#df_ident_cluster=metadata[["Status",args.column]]
#df_ident_cluster.to_csv(os.path.join(args.outdir,"status_cluster.csv"),sep=",",index=False)

File: utils/extract_embedding_cluster.R
library(argparse)
library(stringr)
#library(Seurat)
library(ArchR)
library(Matrix)
library(ggplot2)

#############################
parser <- ArgumentParser(description='Process some tasks')
parser$add_argument("--archr",
                    type="character",
		    default=NULL,
                    help="the path of project saved")

parser$add_argument("--column",
                    type="character",
                    default="Clusters",
		    help="the column in cellcol to be exported")


parser$add_argument("--outdir",
                    type="character",
                    default="ArchR_result")


parser$add_argument("--num_threads",
                    type="integer",
                    default=16,
                    help="number of threads for command")

parser$add_argument("--loupe_cell",
                    action="store_true",
                    default=FALSE)

args <- parser$parse_args()

options(stringsAsFactors=FALSE)
if(!dir.exists(args$outdir)){
        dir.create(args$outdir,recursive=TRUE)
}



print(paste0("Setting threads  :",args$num_threads))
addArchRThreads(threads = args$num_threads) 

print("# Loading data")
proj<-readRDS(args$archr)


print("# Get cellcolData from ArchR Project")
cells=getCellNames(proj)
if(args$loupe_cell){
	cells=proj$Barcode
}
mat=getEmbedding(proj,"TSNE")
colnames(mat)=c("tSNE_1","tSNE_2")
mat$Barcode=cells
mat=mat[,c(3,1,2)]
write.table(mat,file.path(args$outdir,"tsne_projection.csv"),sep=",",quote=F,row.names = F)

mat=getEmbedding(proj,"UMAP")
colnames(mat)=c("UMAP_1","UMAP_2")
mat$Barcode=cells
mat=mat[,c(3,1,2)]
write.table(mat,file.path(args$outdir,"umap_projection.csv"),sep=",",quote=F,row.names = F)

meta=as.data.frame(getCellColData(proj,select=args$column))
meta$barcode=cells
meta=meta[,c(2,1)]
write.table(meta,file.path(args$outdir,"cluster.csv"),sep=",",quote=F,row.names = F)


